src/cluster_legacy.c

```diff
@@ -3613,13 +3625,17 @@ void clusterLinkConnectHandler(connection *conn) {
          * replaced by the clusterSendPing() call. */
         node->ping_sent = old_ping_sent;
     }
-    /* We can clear the flag after the first packet is sent.
-     * If we'll never receive a PONG, we'll never send new packets
-     * to this node. Instead after the PONG is received and we
-     * are no longer in meet/handshake status, we want to send
-     * normal PING packets. */
-    node->flags &= ~CLUSTER_NODE_MEET;
-
+    /* NOTE: We cannot clear the MEET flag from the node until we get a response
```

Comment:
This makes sense to me, but I thought we briefly discussed just only sending the meet once, and on reconnect just not sending another meet. The previously logic was "We'll only send a single meet", I'm wondering if there was any logic somewhere that relied on that behavior.

Reply:
Are we getting into the territory that `CLUSTER MEET` should only be sent when it was generated by the user/admin?

Reply:
The initial MEET is still triggered by an admin/user command.

```
    /* We can clear the flag after the first packet is sent.
     * If we'll never receive a PONG, we'll never send new packets
     * to this node. Instead after the PONG is received and we
     * are no longer in meet/handshake status, we want to send
     * normal PING packets. */
```

The previous code comment above doesn't restrict the MEET to be sent exactly once. I can't think of any scenario where sending multiple MEETs will break - this is similar to when an admin sends MEET multiple times. We send MEET only when the connection is established in `clusterLinkConnectHandler`, so we still limit sending MEET when connection is newly setup and not in every cluster cron run.

Reply:
> this is similar to when an admin sends MEET multiple times

I believe the cluster deduplicate multiple identical requests to the same IP/port.

Reply:
So, is there a worst case scenario where the node which was intended to connect to earlier has changed and we send the MEET command to the wrong node with this change?

Reply:
I can't think of any. I went through the code and it seems okay to get a double MEET. My ask is can we update the test to also cause this edge case. I believe we can drop the pong messages on the source node, kill the connection, and trigger a second meet message.

Reply:
In the test case I am using multiple MEETs to make sure we don't stop dropping the link early. 

```tcl
    wait_for_condition 1000 50 {
        [CI $b cluster_stats_messages_meet_received] >= 3
    } else {
      fail "Cluster node $a never sent multiple MEETs to $b"
    }

```

Reply:
B isn't processing these right? It's just immediately dropping the first three and not processing them, it only ever processes the 4th one once correct?

Reply:
I am not concerned with sending multiple "MEET". It should be idempotent but agreed that we should add a test to validate it (not using the debug hook to fail the previous MEETs)

Reply:
> B isn't processing these right? It's just immediately dropping the first three and not processing them, it only ever processes the 4th one once correct?

Correct. It starts processing when we drop the filter - which can be 4th or later.

> agreed that we should add a test to validate it (not using the debug hook to fail the previous MEETs)

To test this scenario, the test has to send packets to the target node pretending to be a Valkey node. This way we can send multiple MEETs. Can you point me to any test that directly sends cluster messages to nodes, so I can write one with multiple MEETs?

Reply:
I think the test could look like: 
1. Node A is told to meet Node B. We drop all incoming packets on A, so although B get's the response it is never acknowledged.
2. We force B to drop it's connection with A. We support `DEBUG CLUSTERLINK KILL <node id> all`. (I forgot about this till just now).
3. A will reconnect, and send another meet.

Reply:
Thanks! Added test to make sure multiple meets are handled just like a PING.

```
           **MEET**
Node A -----------------> Node B
      X <----------------
             PONG

		   **MEET**
      ------------------>
      X <----------------
             PONG

             PING
      <------------------
      ------------------>
             PONG

		   **MEET**
      ------------------>
      X <----------------
             PONG

      ...

	[Remove DROP PONG]

		   **MEET**
      ------------------>
      <------------------
             PONG

             PING
      ------------------>
      <------------------
             PONG

             PING
      <------------------
      ------------------>
             PONG

```